src/__tests__/api.test.ts

```diff
@@ -35,31 +36,17 @@ jest.mock("https", () => {
 });
 
 jest.mock("https-proxy-agent", () => {
-  return jest
-    .fn()
-    .mockImplementation((opts: httpsProxyAgent.HttpsProxyAgentOptions) => {
-      const agentInstance: httpsProxyAgent.HttpsProxyAgentOptions = {};
+  return {
+    HttpsProxyAgent: jest
+      .fn()
+      .mockImplementation((proxy: URL | string, opts?: ProxyAgentOptions) => {
+        if (!opts || !opts.pfx) {
+          return proxy;
+        }
```

Comment:
IMO: Because the constructor of `HttpsProxyAgent` should return the object instance, it's better to return the object here.

```suggestion
      .mockImplementation(
        (proxy: URL | string, opts?: ProxyAgentOptions): object => {
          if (!opts || !opts.pfx) {
            return { proxy };
          }
```